### ISSUE TITLE
twistd web --wsgi=myapplication sends SCRIPT_NAME='/' and PATH_INFO='/foo/bar' to the application.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3721]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)| @dreid reported|
|-|-|
|Trac ID|trac#3721|
|Type|defect|
|Created|2009-03-30 20:22:59Z|
|Branch|https://github.com/twisted/twisted/tree/wsgi_script_name_root-3721|

This causes urls to be built as //photos/details/1
And this specifically also causes django to sometimes redirect to http://photos/details/1

<details><summary>Searchable metadata</summary>

```
trac-id__3721 3721
type__defect defect
reporter__dreid dreid
priority__normal normal
milestone__ 
branch__branches_wsgi_script_name_root_3721 branches/wsgi_script_name_root-3721
branch_author__dreid dreid
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1238444579000000 1238444579000000
changetime__1238455838000000 1238455838000000
version__None None
owner__dreid dreid

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_1"></a>|@dreid set status to `assigned`|
|-|-|

Robert Brewer says that SCRIPT_NAME could be either, but cherrypy and django's runserver both use an empty string.

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_2"></a>|@dreid commented|
|-|-|

(In [26542]) Branching to 'wsgi_script_name_root-3721'

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_3"></a>|@dreid commented|
|-|-|

(In [26554]) Refs [#3721](https://github.com/twisted/twisted/issues/3721) - Fix the values of SCRIPT_NAME and PATH_INFO in certain cases.

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)|@dreid removed owner<br>@dreid set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)<a name="note_5"></a>|@mwhudson set owner to @dreid|
|-|-|

r=me.

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_6"></a>|@dreid set status to `closed`|
|-|-|

(In [26557]) Treat SCRIPT_NAME and PATH_INFO just like every other WSGI container.

Author: dreid
Reviewer: mwh
Fixes: [#3721](https://github.com/twisted/twisted/issues/3721)
